inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/tasks/SelectionTask.java

```diff
@@ -177,23 +184,56 @@ public void run()
                                 user.getUsername(), recommenderName, score,
                                 threshold);
                     }
-
-                    appEventPublisher.publishEvent(new RecommenderEvaluationResultEvent(this,
-                            recommender, user.getUsername(), result,
-                            System.currentTimeMillis() - start, activated));
+                    publishEvalEvent(user, recommenders.size(), recommenderCount, recommender,
+                            start, result, activated);
+                    
                 }
                 catch (Throwable e) {
-                    log.error("[{}][{}]: Failed", user.getUsername(), recommenderName, e);
+                    log.error("[{}][{}]: Failed", userName, recommenderName, e);
+                    appEventPublisher
+                            .publishEvent(new TaskUpdateEvent(this, userName, TaskState.DONE, 1,
+                                    String.format("Recommender %s failed.", recommender)));
                 }
             }
-    
             recommendationService.setActiveRecommenders(user, layer, activeRecommenders);
         }
 
+     
+        
         schedulingService.enqueue(new TrainingTask(user, getProject(),
                 "SelectionTask after activating recommenders"));
     }
 
+    protected void handleSkippedRecommenderEvals(String userName,
+            List<EvaluatedRecommender> aActiveRecommenders, Recommender aRecommender,
+            String aRecommenderName, String aMessage)
+    {
+        log.debug("[{}][{}]: Activating [{}] without evaluating - {}",
+                userName, aRecommenderName, aRecommenderName, aMessage);
+        aActiveRecommenders.add(
+                new EvaluatedRecommender(aRecommender, EvaluationResult.skipped(), true));
+        appEventPublisher
```

Comment:
Is it correct to send a `TaskState.DONE` here? There is just a `continue` here to proceed with the next recommender - this is not a `break`.